### ISSUE TITLE
The script failed to set correct access policies for key vault

### ISSUE THREAD

**Issue by david-salac** [alan-turing-institute/data-safe-haven#1323]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** MS Win 10
- **Data Safe Haven version:** v4.0.0

## :cactus: Required modules

N/A

## :no_entry_sign: Describe the problem

After running the `Deploy_SHM.ps1` command, it logs the following error.

Do you please know what is wrong? Thank you

**Note:** Step 6 says something about `SRE ID`, but it is not clear whether it is needed for running the `Deploy_SHM.ps1` command nor how to pass it as an argument.
## :deciduous_tree: Log messages

```
> Deploy_SHM.ps1 -shmId mcshdev

...

2022-10-20 10:28:16 [   INFO]: Ensuring that resource group 'RG_SHM_MCSHDEV_SECRETS' exists...
2022-10-20 10:28:16 [   INFO]: [ ] Creating resource group 'RG_SHM_MCSHDEV_SECRETS'
2022-10-20 10:28:17 [SUCCESS]: [✔] Created resource group 'RG_SHM_MCSHDEV_SECRETS'
2022-10-20 10:28:17 [   INFO]: Ensuring that key vault 'kv-shm-mcshdev' exists...
2022-10-20 10:29:04 [   INFO]: [ ] Creating key vault 'kv-shm-mcshdev'
2022-10-20 10:29:38 [SUCCESS]: [✔] Created key vault 'kv-shm-mcshdev'
2022-10-20 10:29:38 [   INFO]: Giving group 'mcshadmin' access to key vault 'kv-shm-mcshdev'...
Set-AzKeyVaultAccessPolicy: Cannot validate argument on parameter 'ObjectId'. The argument is null or empty. Provide an argument that is not null or empty, and then try the
command again.

2022-10-20 10:29:41 [FAILURE]: [x] Failed to set correct access policies for key vault 'kv-shm-mcshdev'!
Exception: Failed to set correct access policies for key vault 'kv-shm-mcshdev'!
```

## :recycle: To reproduce

Described above.

**Comment by jemrobinson** [comment]
What is the output of running the following command in a terminal?

```pwsh
> (Get-AzADGroup -DisplayName "mcshadmin").Id | Select-Object -First 1
```

**Comment by david-salac** [comment]
Weird, but it's empty

**Comment by jemrobinson** [comment]
Have you created a group with this name in the AzureAD tenant where you're deploying the Azure infrastructure?

Note that depending on your setup this may not be the tenant which you created in the [Setup Azure Active Directory (AAD)](https://alan-turing-institute.github.io/data-safe-haven/develop/deployment/deploy_shm.html#setup-azure-active-directory-aad) step.

**Comment by jemrobinson** [comment]
@david-salac : Any updates on this? I've updated the documentation which I think should close this issue. Let me know if you disagree.

**Comment by jemrobinson** [comment]
I've included these fixes in a new release: v4.0.1
